File: 20230102/LongJump/src/main/java/LongJump.java
public class LongJump {
    public long solution(int n) {
        if (n == 1) {
            return 1;
        }

        long[] dp = new long[n];
        dp[0] = 1;
        dp[1] = 2;

        for (int i = 2; i < n; i++)
            dp[i] = (dp[i - 2] + dp[i - 1]) % 1234567;
        
        return dp[n - 1];
    }
}

File: 2023/20230103/LongJump.js
export default function solution(n) {
  const dp = [];
  dp[1] = 1;
  dp[2] = 2;

  for (let i = 3; i <= 2000; i += 1) {
    dp[i] = (dp[i - 1] + dp[i - 2]) % 1234567;
  }

  return dp[n];
}
